### PR TITLE
🚀 [Feature]: add envvar expose middleware

### PR DESCRIPTION
**Pull request by mstrYoda**
## Description

This pr adds environment variable expose middleware.

Fixes https://github.com/gofiber/fiber/issues/2053

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Checklist:

- [X] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation - https://github.com/gofiber/docs for https://docs.gofiber.io/
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] If new dependencies exist, I have checked that they are really necessary and agreed with the maintainers/community (we want to have as few dependencies as possible)
- [X] I tried to make my code as fast as possible with as few allocations as possible
- [ ] For new code I have written benchmarks so that they can be analyzed and improved

### CODE REVIEW

**Comment by efectn** [comment]
Please update awesome-fiber and https://github.com/gofiber/fiber#-internal-middleware

File: middleware/envvar/envvar.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	return func(c *fiber.Ctx) error {
+		path := c.Path()
+		if len(path) < len(cfg.Path) || !strings.HasPrefix(path, cfg.Path) {
+			return c.Next()
+		}
+
+		if path == cfg.Path {
+			envVar := newEnvVar(cfg)
+			varsByte, err := json.Marshal(envVar)
```
Comment (efectn) [inline middleware/envvar/envvar.go:49]: Get this from app config instead of using hardcoded stdlib

**Review by efectn — CHANGES_REQUESTED** [review]

File: middleware/envvar/envvar.go (hunk 1 above)
Comment (mstrYoda) [reply middleware/envvar/envvar.go]: Thx, done 👍

**Comment by mstrYoda** [comment]
> Please update awesome-fiber and https://github.com/gofiber/fiber#-internal-middleware

All done 👍

**Review by efectn — CHANGES_REQUESTED** [review]
Dont forget adding middleware section to other READMEs

**Comment by efectn** [comment]
Check actions please. Lgtm

**Comment by mstrYoda** [comment]
> Check actions please. Lgtm

I changed `t.Setenv` to `os.Setenv` to provide backward compatibility with older Go versions.

**Comment by ReneWerner87** [comment]
please check again the methods of envvar (some of them are not use and lowercase -> so not exposed )

can you please also provide a better example and link where such functionality is used in other languages or frameworks so i can better understand the use of middleware

**Comment by mstrYoda** [comment]
> please check again the methods of envvar (some of them are not use and lowercase -> so not exposed )
> 
> can you please also provide a better example and link where such functionality is used in other languages or frameworks so i can better understand the use of middleware

Well, I quickly searched but could not find an exact example of this. People are trying to reimplement their middleware/components for this work. We always wonder about our env vars that contain feature flags too in our APIs.
This is why I thought that it would be good to have built-in one.

File: middleware/envvar/envvar.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
+	"os"
+	"strings"
+)
+
+// Config defines the config for middleware.
+type Config struct {
+	// Path specifies the path of handler that exposes environment variables
+	Path string
```
Comment (ReneWerner87) [inline middleware/envvar/envvar.go:13]: specifying the path is not necessary in my opinion
you can say outside where the middleware should run

```go
app.Use("/expose/envvars", envvar.New(envvar.Config{
    ...
}))
```

Comment (mstrYoda) [reply middleware/envvar/envvar.go:13]: So we should not support app.Use(envvars.New()) approach? User must provide path?

Comment (efectn) [reply middleware/envvar/envvar.go:13]: Like monitor middleware

**Review by efectn — APPROVED** [review]

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**